src/main/java/com/unigrad/funiverseappservice/controller/GroupController.java

```diff
@@ -27,51 +27,43 @@ public class GroupController {
 
     private final IGroupService groupService;
 
-    private final ICurriculumService curriculumService;
+    private final GroupMemberService groupMemberService;
 
-    private final IUserDetailService userDetailService;
-
-    private final ISyllabusService syllabusService;
-
-    public GroupController(IGroupService groupService, ICurriculumService curriculumService, IUserDetailService userDetailService, ISyllabusService syllabusService) {
+    public GroupController(IGroupService groupService, GroupMemberService groupMemberService) {
         this.groupService = groupService;
-        this.curriculumService = curriculumService;
-        this.userDetailService = userDetailService;
-        this.syllabusService = syllabusService;
+        this.groupMemberService = groupMemberService;
     }
 
     @PostMapping
     public ResponseEntity<Group> createGroup(@RequestBody Group newGroup) {
 
-        Group.Type newGroupType= newGroup.getType();
-
+        Group.Type newGroupType = newGroup.getType();
 
+        switch (newGroupType) {
 
-        switch (newGroupType){
-
-            case CLASS ->{
-                if(newGroup.getName() == null || newGroup.getName().isEmpty()){
+            case CLASS -> {
+                if (newGroup.getName() == null || newGroup.getName().isEmpty()) {
                     throw new MissingRequiredPropertyException();
```

Comment:
Use `StringUtils.isBlank(newGroup.getName())` and `MissingRequiredPropertyException("Name); `